			   READLINE PATCH REPORT
			   =====================

Readline-Release: 8.3
Patch-ID: readline83-006

Bug-Reported-by:	Félix Bouynot <felix.bouynot@setenforce.one> 
Bug-Reference-ID:	<39c9472ddcf23881693380c4b895dd3799b45cec.camel@setenforce.one>
Bug-Reference-URL:	https://lists.gnu.org/archive/html/bug-bash/2026-05/msg00065.html

Bug-Description:

If readline handles a SIGWINCH and resizes its idea of the screen
dimensions, it needs to recompute the columns where the prompt wraps
lines every time, not just when the screen width decreases.

Patch (apply with `patch -p0'):

*** ../readline-8.3-patched/display.c	Thu Sep  4 11:55:37 2025
--- display.c	Fri May 22 14:55:01 2026
***************
*** 3526,3531 ****
      rl_crlf ();
  
!   if (_rl_screenwidth < prompt_visible_length)
!     _rl_reset_prompt ();		/* update local_prompt_newlines array */
  
    /* Redraw only the last line of a multi-line prompt. */
--- 3521,3527 ----
      rl_crlf ();
  
!   /* Let expand_prompt() update local_prompt_newlines and local_prompt_invis_chars
!     arrays */
!   _rl_reset_prompt ();
  
    /* Redraw only the last line of a multi-line prompt. */
*** ../readline-8.3/patchlevel	2013-11-15 08:11:11.000000000 -0500
--- patchlevel	2024-03-21 08:28:40.000000000 -0400
***************
*** 1,3 ****
  # Do not edit -- exists only for use by patch
  
! 5
--- 1,3 ----
  # Do not edit -- exists only for use by patch
  
! 6
